src/praisonai/praisonai/cli/interactive/async_tui.py

```
@@ -262,6 +264,120 @@ def _scan_workspace(self):
+                content=(
+                    "Checkpointing is disabled. Enable with checkpoints.auto: true "
+                    "or PRAISONAI_CHECKPOINTS=on."
+                ),
+            ))
+            return
+        if not ckpt.turns:
+            self.messages.append(ChatMessage(role="system", content="No checkpoints to revert to."))
+            return
+        n = 1
+        token = str(args).strip().lower()
+        if token and token != "last":
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "token"

(S105)